server/routerlicious/packages/lambdas-driver/src/kafka-service/runner.ts

```diff
@@ -51,6 +52,7 @@ export class KafkaRunner implements IRunner {
 		this.factory.on("error", (error) => {
 			this.runnerMetric.error("Kafka factory encountered an error", error);
 			deferred.reject(error);
+			this.deferred = undefined;
```

Comment:
Could you educate me that how does this reset help us here? For the promise cleanup? What the current behavior without this line. Thanks

Reply:
Hey @yunho-microsoft , yes this is to cleanup the promise, so that it is not rejected or resolved again in the stop method. 
As per current behavior, if we dont set this to undefined and reject/resolve again for some reason, it is a no-op, i.e. doesnt do anything. But I think it is better to clean it up after completing.